tests/main/dbus-activation-session-legacy/task.yaml

```diff
@@ -0,0 +1,52 @@
+summary: D-Bus session services support activation without a user of instance systemd
+
+prepare: |
+    snap set system experimental.user-daemons=true
+    snap set system experimental.dbus-activation=true
+
+restore: |
+    systemctl stop private-session-bus.service || true
+    rm -f dbus-launch.env run-session-bus.sh
+    snap unset system experimental.dbus-activation
+    snap unset system experimental.user-daemons
+
+execute: |
+    # This filters out Ubuntu 14.04 and Ubuntu Core respectively
+    if [ "$(command -v systemd-run)" == "" ] || [ "$(command -v dbus-launch)" == "" ]; then
+        exit 0
+    fi
+
+    cat << EOF > run-session-bus.sh
+    #!/bin/sh
+    cd $(pwd)
+    exec dbus-launch --sh-syntax > dbus-launch.env
+    EOF
+    chmod a+x run-session-bus.sh
+
+    echo "Run a private D-Bus session bus as a systemd service"
+    systemd-run --unit=private-session-bus.service \
+        --service-type=forking ./run-session-bus.sh
+    # CentOS 7's systemd does not seem to wait for the appear to wait
+    # for the service to finish starting, so wait for the environment
+    # file to be non-empty
+    retry -n 3 --wait 1 test -s dbus-launch.env
+    eval "$(cat dbus-launch.env)"
+
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+
+    echo "Install a snap containing an activatable D-Bus session service"
+    snap install --edge test-snapd-dbus-service
+
+    echo "A service activation file has been created"
+    [ -f /var/lib/snapd/dbus-1/services/io.snapcraft.SnapDbusService.service ]
+
+    echo "Making a method call wakes the service"
+    install_local test-snapd-dbus-service-client
+    snap connect test-snapd-dbus-service-client:dbus-session-plug \
+                 test-snapd-dbus-service:dbus-session-slot
+    test-snapd-dbus-service-client.session | MATCH hello
```

Comment:
I've changed this to match "hello world".  Changing the string would require rebuilding and publishing the `test-snapd-dbus-service` snap, which I'd prefer to avoid in the short term given it is likely to get stuck requiring manual review again.